scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,94 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments_docstrings(filename, content):
+    """Detects whether comments and docstrings end with a period.
+
+    Args:
+        filename: str. Name of the file.
+        content: str. Contents of the file.
+
+    Returns:
+        bool. True if there is bad pattern else false.
+    """
+    regexp_comment = r"#[^\n]*[^.,\n]\n( |\t)*(#(( |\t)*[A-Z])|[^#\s])"
+    message_comment = "There should be a period after comment end."
+    message_docstring = "There should be a period after docstring end."
+    punctuate = ['.',',',';',':']
+    excluded_words = ['utf','pylint:','http://']
+    excluded_dirs = ()
+    excluded_files = ()
+    if not (any(filename.startswith(excluded_dir)
+                for excluded_dir in excluded_dirs)
+            or filename in excluded_files):
+        bad_pattern_count = 0
+        # Finds comments with a missing period.
+        for matchobj in re.finditer(regexp_comment, content):
+            start = matchobj.start()+1
+            line_num = find_line(content, start)
+            if not any(word in content.splitlines(True)[line_num-1] for word in excluded_words):
+                print '%s --> Line %s: %s' % (
+                    filename, line_num, message_comment)
+                bad_pattern_count += 1
+        # Finds docstrings with a missing period.
```

Comment:
It might be worth pulling this section out into a separate function since it's quite long.